### ISSUE TITLE
Profile Federation Issues

### ISSUE THREAD

**Issue by justindthomas** [pixelfed/pixelfed#3709]
I just installed Pixelfed on my own instance ( https://pixels.jdt.io ). After a little bit of tweaking, I think it's installed properly (i.e., the [Diagnostics dashboard output](https://gist.github.com/justindthomas/0766e877e903ea699c8e825601a1dc8f) looks like I'd expect).

I'm having some trouble accessing other folks' profiles via federation. Using the search function in the current interface (apparently, it's "new" I guess?) doesn't work at all. In the nginx logs, it generates a 500:

```
10.42.128.22 - - [21/Oct/2022:18:39:34 +0000] \
"GET /api/v2/search?q=https:%2F%2Fpixelfed.social%2Fdansup&resolve=true HTTP/1.1" 500 103 \
"https://pixels.jdt.io/i/web/profile/488790428112244737" "Mozilla/5.0 (X11; Linux x86_64; rv:105.0) \
Gecko/20100101 Firefox/105.0"
```

Attempting it with the `@name@instance` format also fails:

```
10.42.128.22 - - [21/Oct/2022:18:47:23 +0000] \
"GET /api/v2/search?q=%40dansup%40pixelfed.social&resolve=true HTTP/1.1" 500 103 \
"https://pixels.jdt.io/i/web" "Mozilla/5.0 (X11; Linux x86_64; rv:105.0) \
Gecko/20100101 Firefox/105.0"
```
If I switch to the old interface, using the link in my profile, I can search there and I can see the profile:

![image](https://user-images.githubusercontent.com/354734/197267333-4d764100-e18b-44e8-85eb-784cb8a80314.png)

But when I select it, I only see a blank screen:

![image](https://user-images.githubusercontent.com/354734/197267510-d75a480b-0076-495f-81f8-ad791c0a7132.png)

I'm very unfamiliar with the PHP ecosystem, so I'm not sure where to look for application logs. I did enable the "APP_DEBUG" configuration item, but I haven't figured out where that actually gets written to - any advice there to help me troubleshoot this issue would be appreciated.

Fwiw, I was able to follow my Pixelfed user from my Mastodon instance ( https://ser.endipito.us ) and I received the notification of the follow in Pixelfed. So it seems that ActivityPub is working in general. It just doesn't seem to be pulling the Profiles down for some reason (and the search function in the new interface just seems broken).

**Comment by pageuppagedown** [comment]
I'm experiencing the same problem.

* The search in the new interface never returns any results. The spinner spins indefinitely.
* The old interface returns a profile, but nothing comes up when I click on View Profile. This happens when attempting to follow `@dansup@pixelfed.social` from my own Pixelfed instance (`netmonkey.pictures`).
* I can confirm that it returns a blank page when attempting to view the profile of someone on a less busy instance, such as `pixel.artemai.art`.
* I can also confirm that ActivityPub is working on my instance.
* I'm using `v0.11.4` from the `dev` branch.

**Comment by justindthomas** [comment]
Just updated with the latest commits and the problem still exists. Seems to be in the `relationships` endpoint.
![Screenshot from 2022-11-07 20-08-40](https://user-images.githubusercontent.com/354734/200473098-f3f6e863-2494-4ef9-a27a-c2e8696b005a.png)

**Comment by pageuppagedown** [comment]
This is the error that's being returned in the HTTP 500 response when querying `@dansup@pixelfed.social` in the new user interface.

```
{
    "error": "Unable to read key from file file:///srv/http/pixelfed/storage/oauth-public.key"
}
```

I can confirm that `oauth-public.key` does _not_ exist on my instance. This error is _also_ the reason that I'm unable to reply to any comments. I'm going to spend some time researching this.

**Comment by pageuppagedown** [comment]
@justindthomas Once I figured out that this was an OAuth problem, I was able to find troubleshooting documentation very easily.

https://docs.pixelfed.org/running-pixelfed/troubleshooting/#apps-using-oauth-login-show-something-went-wrong-page

I'm not sure if this is something that's missing from the Arch Linux instructions or not (will check and add it to my notes), but all I had to do was execute `php artisan passport:install` and it created the OAuth keypair in the `storage` directory. I'm now able to search and follow `@dansup@pixelfed.social` from my Pixelfed instance and I can like and reply to comments. All is well for me now. Yay!
